public/docs/ts/latest/guide/testing.jade

```diff
@@ -1,102 +1,1758 @@
+  chained in a long sequence of promise callbacks.
+
+a#jasmine-done
+:marked
+  ## _jasmine.done_
+
+  While `fakeAsync` and even `async` function greatly simplify Angular asynchronous testing,
+  you can still fallback to the traditional Jasmine asynchronous testing technique.
+
+  You can still pass `it` a function that takes a 
+  [`done` callback](http://jasmine.github.io/2.0/introduction.html#section-Asynchronous_Support).
+  Now you are responsible for chaining promises, handling errors, and calling `done` at the appropriate moment.
+
+  Here is a `done` version of the previous two tests:
++makeExample('testing/ts/app/shared/twain.component.spec.ts', 'done-test', 'app/shared/twain.component.spec.ts (done test)')(format='.')
+:marked
+  Although we have no direct access to the `getQuote` promise inside `TwainComponent`,
+  the spy does and that makes it possible to wait for `getQuote` to finish.
+  
+  The `jasmine.done` technique, while discouraged, may become necessary when neither `async` nor `fakeAsync`
+  can tolerate a particular asynchronous activity. That's rare but it happens.
+
+a(href="#top").to-top Back to top
+
+.l-hr
+
+a#component-with-external-template
+:marked
+  # Test a component with an external template
+  The `TestBed.createComponent` is a synchronous method. 
+  It assumes that everything it could need is already in memory.
+
+  That has been true so far. 
+  Each tested component's `@Component` metadata has a `template` property specifying an _inline templates_.
+  Neither component had a `styleUrls` property.
+  Everything necessary to compile them was in memory at test runtime.
+
+  The `DashboardHeroComponent` is different. 
+  It has an external template and external css file, specified in `templateUrl` and `styleUrls` properties.
++makeExample('testing/ts/app/dashboard/dashboard-hero.component.ts', 'component', 'app/dashboard/dashboard-hero.component.ts (component)')(format='.')
+:marked
+  The compiler must read these files from a file system before it can create a component instance.
+
+  The `TestBed.compileComponents` method asynchronously compiles all the components configured in its
+  current test module. After it completes, external templates and css files, have been "inlined"
+  and `TestBed.createComponent` can do its job synchronously.
+
+  The `app/dashboard/dashboard-hero.component.spec.ts` demonstrates the pre-compilation process:
++makeExample('testing/ts/app/dashboard/dashboard-hero.component.spec.ts', 'compile-components', 'app/dashboard/dashboard-hero.component.spec.ts (compileComponents)')(format='.')
+
+a#async-fn-in-before-each
+:marked
+  ## The _async_ function in _beforeEach_
+
+  Notice the `async` call in the `beforeEach`.
+
+  The `async` function is part of the _Angular TestBed_ feature set.
+  It _takes_ a parameterless function and _returns_ a parameterless function
+  which becomes the argument to the Jasmine `beforeEach` call. 
```

Comment:
Yes ... I explained in the context of `it`.  I deliberately repeated here in the context of `beforeEach` because I couldn't be sure you would see it here first or there first. Keeping for now.